client/src/partials/profit_center/center/analysis_profit_center.js

```diff
@@ -1,99 +1,106 @@
-      $scope.selected = angular.copy(profit_center);
-      removeProfitCenter()
-      .then(function () {
-        $scope.model.profit_centers.remove($scope.selected.id);
-        messenger.success($translate.instant('ANALYSIS_PROFIT_CENTER.REMOVE_SUCCESS_MESSAGE'));
-      })
-      .catch(function () {
-        messenger.danger($translate.instant('ANALYSIS_PROFIT_CENTER.REMOVE_FAIL_MESSAGE'));
+  function save() {
+    vm.register.project_id = vm.project.id;
+    writeCenter()
+    .then(function() {
+      validate.refresh(dependencies, ['profit_centers']).then(function (model) {
+        angular.extend(vm, model);
       });
-    }
+      vm.register = {};
+      vm.action = 'default';
+      messenger.success($translate.instant('ANALYSIS_PROFIT_CENTER.INSERT_SUCCESS_MESSAGE'));
+    })
+    .catch(function () {
+      messenger.danger($translate.instant('ANALYSIS_PROFIT_CENTER.INSERT_FAIL_MESSAGE'));
+    });
+  }
 
-    function edit() {
-      delete $scope.selected.abbr;
-      updateProfitCenter()
-      .then(function () {
-        $scope.model.profit_centers.put($scope.selected);
-        messenger.success($translate.instant('ANALYSIS_PROFIT_CENTER.UPDATE_SUCCESS_MESSAGE'));
-      })
-      .catch(function () {
-        messenger.danger($translate.instant('ANALYSIS_PROFIT_CENTER.UPDATE_FAIL_MESSAGE'));
-      });
-    }
+  function remove(profit_center) {
+    vm.selected = angular.copy(profit_center);
+    removeProfitCenter()
+    .then(function () {
+      vm.model.profit_centers.remove(vm.selected.id);
+      messenger.success($translate.instant('ANALYSIS_PROFIT_CENTER.REMOVE_SUCCESS_MESSAGE'));
+    })
+    .catch(function () {
+      var msg = $translate.instant('ANALYSIS_PROFIT_CENTER.REMOVE_FAIL_MESSAGE').replace('%VAR%', vm.selected.text);
+      messenger.error(msg);
+    });
+  }
 
-    function removeProfitCenter() {
-      return connect.delete('profit_center', 'id', [$scope.selected.id]);
-    }
+  function edit() {
+    delete vm.selected.abbr;
+    updateProfitCenter()
+    .then(function () {
+      vm.model.profit_centers.put(vm.selected);
+      vm.action = 'default';
+      messenger.success($translate.instant('ANALYSIS_PROFIT_CENTER.UPDATE_SUCCESS_MESSAGE'));
+    })
+    .catch(function () {
+      messenger.error($translate.instant('ANALYSIS_PROFIT_CENTER.UPDATE_FAIL_MESSAGE'));
+    });
+  }
 
-    function updateProfitCenter() {
-      return connect.put('profit_center', [connect.clean($scope.selected)], ['id']);
-    }
+  function removeProfitCenter() {
+    return connect.delete('profit_center', 'id', [vm.selected.id]);
+  }
 
-    $scope.setAction = setAction;
-    $scope.saveRegistration = saveRegistration;
-    $scope.remove = remove;
-    $scope.edit = edit;
+  function updateProfitCenter() {
+    return connect.put('profit_center', [connect.clean(vm.selected)], ['id']);
   }
-]);
+
+  vm.setAction = setAction;
+  vm.save = save;
+  vm.remove = remove;
+  vm.edit = edit;
```

Comment:
It's helpful to put these bindings at the top of your controller.  That way, we can automatically see what is bound to the view in the first lines of the controller.